### PR TITLE
Support for recognizing complex MIME definitions in OAS3 content type

### LINKED ISSUE #190: Support example for proper json MIME types
Currently only examples under `application/json` are supported. 
However a proper json payload can be indicated by e.g. `application/json;charset=utf-8`, `application/vnd.some.model+json`.
Both examples provided confirm to MIME grammar (https://en.wikipedia.org/wiki/Media_type).

I thought simple improvement of a pattern could help. Like:
`$..responses..content[?(@property.match(/[\/+]json/))].example`.
However, it works only partially due to the how jsonpath-plus parser works internally - in short it uses `;` as fragment separator.
So I could identify all examples with improved json-path but then identification of elements would not work. 

Switching to json pointers could be a way forward.

### PR DESCRIPTION
**Pull request by bartoszm**
fixes #190.

I did not remove escape unescape methods yet but right now they have no use.
Please let me know if you want me to remove them.

### CODE REVIEW

File: test/data/v3/simple-api-with-example-and-examples-mime-complex.json
Comment (codekie) [inline test/data/v3/simple-api-with-example-and-examples-mime-complex.json:1]: Please rename the file to `simple-api-with-example-and-examples-mime-complex-invalid.json`

**Review by codekie — CHANGES_REQUESTED** [review]
Aside from a few minor issues it looks good. Well done 👏 . Thanks a lot for your effort!

File: src/index.js [hunk 1]
```diff
@@ -516,20 +539,16 @@ function _initStatistics() {
...
 /**
- * Extract object(s) by the given JSON-path
- * @param {String}  path    JSON-path
- * @param {Object}  json    JSON to extract the object(s) from
- * @returns {Object|Array.<Object>|undefined} All matching objects. Single object if there is only one match
- * @private
+ * Extract object by the given JSON-pointer
+ * @param {String} pointer JSON-pointer
+ * @param {Object} json JSON to extract the object(s) from
  */
```
[the remark this answers is not in the document]
Comment (bartoszm) [reply src/index.js:541]: fixed

File: src/index.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const
     merge = require('lodash.merge'),
     flatten = require('lodash.flatten'),
     flatMap = require('lodash.flatmap'),
+    jp = require('json-pointer'),
```
[the remark this answers is not in the document]
Comment (bartoszm) [reply src/index.js]: done

File: src/impl/v2/index.js
[the remark this answers is not in the document]
Comment (bartoszm) [reply src/impl/v2/index.js:1]: done

**Review by codekie — APPROVED** [review]

**Comment by codekie** [comment]
Looks good now. Thank you for your contribution! 👏 🎉

**Pull request merged**